builder/hcloud/step_create_server.go

```diff
@@ -242,3 +271,29 @@ func getImageWithSelectors(ctx context.Context, client *hcloud.Client, c *Config
 
 	return allImages[0], nil
 }
+
+func getPrimaryIP(ctx context.Context, client *hcloud.Client, publicIP string) (*hcloud.PrimaryIP, string, error) {
+	var hcloudPublicIP *hcloud.PrimaryIP
+	publicIPID, err := strconv.ParseInt(publicIP, 10, 64)
+	if err == nil {
+		hcloudPublicIP, _, err = client.PrimaryIP.GetByID(ctx, publicIPID)
+		if err != nil {
+			return nil, fmt.Sprintf("Could not fetch primary ip with ID %d", publicIPID), err
+		}
+	} else {
+		hcloudPublicIP, _, err = client.PrimaryIP.GetByName(ctx, publicIP)
+		if err != nil {
+			return nil, fmt.Sprintf("Could not fetch primary ip '%s'", publicIP), err
+		}
```

Comment:
GetByID and GetByName can be replaced with Get. See https://pkg.go.dev/github.com/hetznercloud/hcloud-go/v2/hcloud#PrimaryIPClient.Get